### PR TITLE
add migration functional after register new user

### PR DESCRIPTION
**Pull request by Eroslaev**

### CODE REVIEW

File: admin/class-virgil-pure-admin.php [hunk 1]
```diff
@@ -266,6 +267,10 @@ public function virgil_pure_profile_update(int $userId): void
         $wpUser = get_user_by('id', $userId);
         $this->updatePassword($wpUser);
     }
+    public function virgil_pure_user_register(int $userId): void
+    {
+        $this->fh->migrate();
```
Comment (SergeySeroshtan) [inline admin/class-virgil-pure-admin.php]: Add method: `migrateOneUser($userId)`, like this one - https://github.com/VirgilSecurity/virgil-pure-wordpress/blob/a0d852a528b0660836e5c1a028b4d86b2409fc1c/admin/core/Core/FormHandler.php#L161

**Review by SergeySeroshtan — COMMENTED** [review]
Please address the comment and rename PR to reflect changes.

File: admin/class-virgil-pure-admin.php (hunk 1 above)
Comment (Eroslaev) [reply admin/class-virgil-pure-admin.php]: @SergeySeroshtan added and changed to new method

File: admin/class-virgil-pure-admin.php [hunk 2]
```diff
@@ -266,6 +266,21 @@ public function virgil_pure_profile_update(int $userId): void
...
+                update_option(Option::LAST_CHECK, time());
+                return;
+            }
+
+            if (time() - get_option(Option::LAST_CHECK) > 300) {
+                $this->fh->migrate();
+                update_option(Option::LAST_CHECK, time());
+            }
+        }
+    }
```
Comment (SergeySeroshtan) [inline admin/class-virgil-pure-admin.php]: This hook is not needed.
Instead create a background job that will fetch non-migrated users each N minutes and migrate them.

File: admin/core/Background/EncryptAndMigrateBackgroundProcess.php [hunk 3]
```diff
@@ -133,15 +132,26 @@ protected function task(mixed $item): bool
...
+            update_option(Option::AUTO_MIGRATION, $this->identifier);
+
+            $newUsers = $this->dbqh->getNewUsers();
+
+            if (count($newUsers) > 0) {
+                $this->fh->migrate();
+            } else {
+                update_option(Option::LAST_CHECK, time());
+                parent::complete();
+            }
```
Comment (SergeySeroshtan) [inline admin/core/Background/EncryptAndMigrateBackgroundProcess.php]: Next lines can be removed when functionality will be moved to the background process.

**Comment by SergeySeroshtan** [comment]
Superseded by https://github.com/VirgilSecurity/virgil-pure-wordpress/pull/16

**Pull request closed without merging**